server/src/main/java/io/crate/statistics/ReservoirSampler.java

```diff
@@ -213,8 +243,16 @@ private Samples getSamples(List<Reference> columns,
             int readerId = FetchId.decodeReaderId(fetchId);
             DocIdToRow docIdToRow = docIdToRowsFunctionPerReader.get(readerId);
             Object[] row = docIdToRow.apply(FetchId.decodeDocId(fetchId));
+
             try {
-                rowAccounting.accountForAndMaybeBreak(row);
+                long bytesRead = rowAccounting.accountRowBytes(row);
+                ramAccounting.addBytes(bytesRead);
+                totalSizeInBytes += bytesRead;
+                long minPauseCheckBytes = rateLimiter.getMinPauseCheckBytes(); // volatile read, gets updated together with rateLimiter.mbPerSec
+                if (minPauseCheckBytes > 0) {
+                    // Throttling is enabled
+                    bytesSinceLastPause = maybePause(bytesRead, bytesSinceLastPause);
+                }
```

Comment:
minor, but I'd prefer all maybe pause logic to be at one place, so I'd suggest to move that inside `maybePause`.

Reply:
moved in https://github.com/crate/crate/pull/13046/commits/94e60d3014fc2ba2bf6cb5a4c9a5021e55465fae